[Bug]: `/list` での `search` クエリでパースに失敗している

### 追加連絡先

_No response_

### 発生した不具合

`search` クエリの仕様変更に伴い、パースに失敗している（データの取得自体は成功している。）



### 期待した挙動

正常にパースし、データを取得できる

### Wrapperバージョン

v6.0.0.0-Alpha.1

### Relevant log output

```shell
2023-11-24 22:47:07,372 DEBUG [JDA MainWS-ReadThread] h.i.b.ListEndPointBuilder [ListEndPointBuilder.java : 191] set parameters: {search=%E3%81%8A%E9%A1%98%E3%81%84, music_type=cg, limit=1, type=music}
2023-11-24 22:47:07,372 INFO [JDA MainWS-ReadThread] h.i.b.ListEndPointBuilder [ListEndPointBuilder.java : 192] constructing action instance...
2023-11-24 22:47:07,373 DEBUG [JDA MainWS-ReadThread] h.i.b.ListEndPointBuilder [ListEndPointBuilder.java : 195] complete. information: hajimeapi4j.internal.request.ArrayResponseRestActionImpl@4df922a0
2023-11-24 22:47:07,770 DEBUG [JDA RateLimit-Elastic-Worker 1] n.d.j.i.r.Requester [Requester.java : 224] Received response with following cf-rays: [82b21a0e7d10afc3-NRT]
2023-11-24 22:47:08,375 DEBUG [ForkJoinPool.commonPool-worker-2] h.i.r.AbstractRestAction [AbstractRestAction.java : 76] Requesting to https://api.fujiwarahaji.me/v3/list?search=%E3%81%8A%E9%A1%98%E3%81%84&music_type=cg&limit=1&type=music
2023-11-24 22:47:09,291 WARN [MochiMochiTalk Song detail integration thread-Worker 2] M.u.ConcurrencyUtil [ConcurrencyUtil.java : 21] There was an error while handling an event. ret=null, t={}
java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot invoke "java.lang.Class.isPrimitive()" because "clz" is null
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1770)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.lang.NullPointerException: Cannot invoke "java.lang.Class.isPrimitive()" because "clz" is null
	at com.fasterxml.jackson.databind.type.TypeFactory._findWellKnownSimple(TypeFactory.java:1421)
	at com.fasterxml.jackson.databind.type.TypeFactory._fromClass(TypeFactory.java:1508)
	at com.fasterxml.jackson.databind.type.TypeFactory.constructCollectionLikeType(TypeFactory.java:961)
	at hajimeapi4j.internal.request.ArrayResponseRestActionImpl.handleResponse(ArrayResponseRestActionImpl.java:32)
	at hajimeapi4j.internal.request.ArrayResponseRestActionImpl.handleResponse(ArrayResponseRestActionImpl.java:13)
	at hajimeapi4j.internal.request.AbstractRestAction.lambda$handleAsync$0(AbstractRestAction.java:38)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	... 6 common frames omitted
```

--- a/.github/ISSUE_TEMPLATE/bug_report.yaml
+++ b/.github/ISSUE_TEMPLATE/bug_report.yaml
@@ -42,6 +42,8 @@ body:
         - v4.0.0
         - v5.0.0.0
         - v6.0.0.0-Alpha.1
+        - v6.0.0.0
+        - v6.0.1.0
     validations:
       required: true
   - type: textarea

--- a/.github/ISSUE_TEMPLATE/feature_req.yaml
+++ b/.github/ISSUE_TEMPLATE/feature_req.yaml
@@ -49,6 +49,8 @@ body:
         - v4.0.0
         - v5.0.0.0
         - v6.0.0.0-Alpha.1
+        - v6.0.0.0
+        - v6.0.1.0
     validations:
       required: true
   - type: textarea

--- a/lib/build.gradle
+++ b/lib/build.gradle
@@ -72,7 +72,7 @@ tasks.named('javadoc') {
 sourceCompatibility = '11'
 targetCompatibility = '11'
 
-version = 'v6.0.0.0'
+version = 'v6.0.1.0'
 jar.archiveBaseName = 'hajimeapi4j'
 
 test {

--- a/lib/src/main/java/hajimeapi4j/internal/request/ArrayResponseRestActionImpl.java
+++ b/lib/src/main/java/hajimeapi4j/internal/request/ArrayResponseRestActionImpl.java
@@ -1,6 +1,6 @@
 package hajimeapi4j.internal.request;
 
-import com.fasterxml.jackson.core.type.TypeReference;
+import hajimeapi4j.api.endpoint.ListEndPoint;
 import java.io.IOException;
 import java.util.List;
 import java.util.Map;
@@ -13,12 +13,8 @@ import org.jetbrains.annotations.Nullable;
 @Slf4j
 public class ArrayResponseRestActionImpl<T> extends RestActionImpl<List<T>> {
 
-  protected final TypeReference<List<T>> typeRef;
-
   public ArrayResponseRestActionImpl(@Nonnull String pathParam, Map<String, String> queryParam) {
     super(pathParam, queryParam);
-    this.typeRef = new TypeReference<>() {
-    };
   }
 
   @Nullable
@@ -29,7 +25,7 @@ public class ArrayResponseRestActionImpl<T> extends RestActionImpl<List<T>> {
     }
     try {
       return MAPPER.readValue(r.body().string(),
-          MAPPER.getTypeFactory().constructCollectionLikeType(List.class, clazz));
+        MAPPER.getTypeFactory().constructCollectionLikeType(List.class, ListEndPoint.class));
     } catch (IOException e) {
       log.error("An error occurred while handling the response.", e);
     }

--- a/lib/src/test/java/hajimeapi4j/e2e/SearchQueryTest.java
+++ b/lib/src/test/java/hajimeapi4j/e2e/SearchQueryTest.java
@@ -1,0 +1,30 @@
+package hajimeapi4j.e2e;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import hajimeapi4j.api.endpoint.ListEndPoint;
+import hajimeapi4j.internal.builder.ListEndPointBuilder;
+import hajimeapi4j.util.enums.ListParameter.MusicType;
+import hajimeapi4j.util.enums.ListParameter.Type;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+@Disabled("To prevent from excessive API calls")
+class SearchQueryTest {
+
+  @Test
+  void testSearchQuery() {
+    ListEndPointBuilder builder = ListEndPointBuilder.createFor(Type.MUSIC);
+    builder.setLimit(1).setMusicType(MusicType.CINDERELLA_GIRLS).setSearch("お願い");
+
+    builder.build().handleAsync((result) -> {
+      assertEquals(result.size(), 1);
+      ListEndPoint d = result.get(0);
+      assertEquals(d.getType(), Type.MUSIC.toString());
+      assertEquals(d.getMusicType().orElse("unknown"), MusicType.CINDERELLA_GIRLS.toString());
+    }).join();
+  }
+}

